### ISSUE TITLE
🐛 Bug: Field in Table disappeared

### ISSUE THREAD

**Issue by chris2k20** [nocodb/nocodb#9524]
### Please confirm that the bug report does ***not*** already exist

- [X] I confirm there is no existing issue for this bug.

### Steps to reproduce

1. Create a new Table
2. Create a Start Field (Date-Time)
3. Create a End Field (Date-Time) 
4. Create a Formula with `HOUR(DATETIME_DIFF({Start},{End}))`
5. Click "Save changes"
6. The "Formula" created in the fields row as "New field"
7. Click "Save changes" (twice)
![image](https://github.com/user-attachments/assets/e0eb7738-0a44-4e39-9d70-0d8e30beb8be)

8. Reload page
![image](https://github.com/user-attachments/assets/6d6fbcad-2ac4-4621-84a8-166aeb58ec94)

9. Field "Formula" is gone 👎🏻 

### Desired Behavior

That the new field "Formular" stays or I get an error message. 

### Project Details

Using docker image `:0.255.2` in Kubernetes (k3s). 
Postgresql Database (Bitnami) latest Version.

**Comment by o1lab** [comment]
Ok, we are taking a look at this.

**Comment by rameshmane7218** [comment]
Your message could be improved slightly for clarity and grammar:

"Hey @chris2k20, after debugging, I found that `HOUR(DATETIME_DIFF({Start},{End}))` is invalid because `HOUR` requires a `datetime`, but `DATETIME_DIFF` returns an integer. In such cases, we should show a warning instead, and we are fixing it.

**Comment by chris2k20** [comment]
Yeah thx @rameshmane7218 with that `DATETIME_DIFF({Start}, {End}, "hours")` it is working. But as you said a warning would be nice!!!
